drivers/include/mtd.h

```diff
@@ -101,6 +101,27 @@ struct mtd_desc {
                 uint32_t addr,
                 uint32_t size);
 
+    /**
+     * @brief   Read from the Memory Technology Device (MTD) using
+     *          pagewise addressing.
+     *
+     * @p offset should not exceed the page size
+     *
+     * @param[in]  dev      Pointer to the selected driver
+     * @param[out] buff     Pointer to the data buffer to store read data
+     * @param[in]  page     Starting page address
+     * @param[in]  offset   Byte offset from the start of the page
+     * @param[in]  size     Number of bytes
+     *
+     * @return 0 on success
+     * @return < 0 value on error
+     */
+    int (*read_page)(mtd_dev_t *dev,
+                void *buff,
+                uint32_t page,
+                uint16_t offset,
```

Comment:
`page_size` is `uint32_t`. For consistency, this should IMO also be `uint32_t`.

I'm not a storage expert and missing the qualification reason if 16 bit are wide enough. However, a short search seems to indicate that SSDs typically have block sizes 256 KiB and 4 MiB (each consisting typically of 128 or 256 pages with each page sized 2 KiB, 4 KiB, 8 KiB, or 16 KiB). If someone would want to add an SSD driver that maps an MTD page to an SSD block, this would require an `uint32_t` offset. But again: I have no idea if this would make sense to map the MTD page to the SSD block rather than to the SSD page.

Still: With current hardware already having 16KiB pages, maybe it will not take too long until pages > 64KiB are used somewhere.  So I would say it might make sense to use `uint32_t` not just for consistency, but also to future proof the API.